src/frame/vulkan.rs

```diff
@@ -292,41 +314,48 @@ impl Vulkan {
             .usage(vk::ImageUsageFlags::TRANSFER_SRC)
             .sharing_mode(vk::SharingMode::EXCLUSIVE);
 
-        let frame_image = unsafe { self.device.create_image(&frame_image_create_info, None)? };
+        let frame_image = unsafe {
+            self.device
+                .create_image(&frame_image_create_info, None)
+                .expect("Failed to create image")
+        };
 
+        // Memory requirements info
         let frame_image_memory_req_info =
-            vk::ImageMemoryRequirementsInfo2::builder().image(frame_image);
+            vk::ImageMemoryRequirementsInfo2::default().image(frame_image);
 
-        // Prepare the structures to get memory requirements into, then get the requirements
+        // Memory requirements
         let mut frame_image_mem_dedicated_req = vk::MemoryDedicatedRequirements::default();
-        let mut frame_image_mem_req = vk::MemoryRequirements2::builder()
-            .push_next(&mut frame_image_mem_dedicated_req)
-            .build();
+
+        let mut frame_image_mem_req =
+            vk::MemoryRequirements2::default().push_next(&mut frame_image_mem_dedicated_req);
+
         unsafe {
             self.device.get_image_memory_requirements2(
                 &frame_image_memory_req_info,
                 &mut frame_image_mem_req,
             );
         }
 
-        // Bit i in memory_type_bits is set if the ith memory type in the
-        // VkPhysicalDeviceMemoryProperties structure is supported for the image memory.
-        // We just use the first type supported (from least significant bit's side)
+        // Find suitable memory type index
```

Comment:
That comment seems useful even if I don't remember the defaults :sweat_smile: Lets keep it :pray: